protocol/src/format.rs

```diff
@@ -12,22 +12,22 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::fmt;
-use std::mem;
+use std::{fmt, mem, str::FromStr};
 
 use crate::translate;
 use crate::translate::*;
 
 use serde::Deserialize;
+
+pub use crate::format::color::*;
+
 #[derive(Debug, Clone)]
 pub enum Component {
     Text(TextComponent),
 }
 
 impl Component {
     pub fn from_string(str: &str) -> Self {
-        // TODO: Support: §l, §k, §m, §o, §n, §r
```

Comment:
Did you already add support for all of these modifiers or why did you remove this comment?

Reply:
> Did you already add support for all of these modifiers or why did you remove this comment?

Maybe I should add the comment back or phase if differently.
because the modifiers get set. it just don't get rendered.
or maybe I should add the comment to where the modifiers get read?

Reply:
I think you should add it to where they get added or where they are stored and rephrase it a bit